### PR TITLE
Build and Push Docker Image

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing
### Description

Adds a docker file and a few docker compose files for various situations. 

A lot of the seed files are "incomplete" and the docker image won't work without setting them first. Documentation is updated to make it clear how to get up and running.

Also includes a new CI step to build a new docker image for each commit. 

### TODO
- [x] does the ci build overwrite the image? they should be pushed to a commit hash unless merging into dev
- [ ] get correct docker token into ci

### CODE REVIEW

**Comment by miles-grant-ibigroup** [comment]
This is unblocked and ready for review @daniel-heppner-ibigroup

File: docker-compose.dev.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+version: '3.8'
+services:
+  mongo:
+    volumes:
+      - type: bind
+        source: /opt/homebrew/var/mongodb
```
Comment (daniel-heppner-ibigroup) [inline docker-compose.dev.yml:6]: This path doesn't exist on my computer, and I have mongo-community installed with homebrew. Improve docs with required setup

File: docker-compose.e2e.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+version: '3.8'
```
Comment (daniel-heppner-ibigroup) [inline docker-compose.e2e.yml:1]: Explain somewhere that all this file does is seed the database, doesn't actually run e2e tests.

File: README.md [hunk 3]
```diff
@@ -16,7 +16,21 @@ otp-middleware is a potential way for agencies to fill in those gaps. It provide
 - and other features (TODO).
 
 ## Deployment
-TODO - One-time deployment.
+The easiest way to deploy otp-middleware is to use the included Docker Compose configuration.
+
+There are three compose configs. All build upon the "standard" config. The `dev` config exposes the Mongo instance, and saves the database data outside of the container. The `e2e` configuration also exposes the Mongo instance, and pre-seeds the database using the files in the `configurations/e2e-mongo-seed` directory. Please note that the seed files do not work out of the box. Fields which require input are clearly marked.
```
Comment (daniel-heppner-ibigroup) [inline README.md]: - Explain what `dev` config does a bit more, like that you need mongo set up locally for it to work. 
- What does "seed files do not work out of the box" mean?

File: README.md [hunk 4]
```diff
@@ -16,7 +16,21 @@ otp-middleware is a potential way for agencies to fill in those gaps. It provide
...
+
+##### Running otp-middleware via Docker Compose
+```bash
+docker compose -f docker-compose.yml up
+```
+
+##### Running otp-middleware in dev mode via Docker Compose
+```bash
+docker compose -f docker-compose.yml -f docker-compose.dev.yml up
+```
```
Comment (daniel-heppner-ibigroup) [inline README.md:33]: Add e2e mode instructions

**Review by daniel-heppner-ibigroup — APPROVED** [review]

File: docker-compose.dev.yml (hunk 1 above)
Comment (miles-grant-ibigroup) [reply docker-compose.dev.yml:6]: Documentation is expanded in 82a26d9f577223563e97cae49e378197a4196eb8

File: docker-compose.e2e.yml (hunk 2 above)
Comment (miles-grant-ibigroup) [reply docker-compose.e2e.yml:1]: Documentation is expanded in https://github.com/ibi-group/otp-middleware/commit/82a26d9f577223563e97cae49e378197a4196eb8

File: README.md (hunk 3 above)
Comment (miles-grant-ibigroup) [reply README.md]: Documentation is expanded in https://github.com/ibi-group/otp-middleware/commit/82a26d9f577223563e97cae49e378197a4196eb8

File: README.md (hunk 4 above)
Comment (miles-grant-ibigroup) [reply README.md:33]: Documentation is expanded in https://github.com/ibi-group/otp-middleware/commit/82a26d9f577223563e97cae49e378197a4196eb8

**Review by philip-cline — APPROVED** [review]

**Pull request merged**